script/echoclient.py

```diff
@@ -0,0 +1,38 @@
+import socket
+import sys
+
+ser_port = 1234
+cli_port = 4321
+
+def client(ip):
+    server_address = (ip, ser_port)
+    sock = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_TCP
+    )
+    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    sock.bind(('', cli_port))
```

Comment:
you shouldn't need bind() for the client socket. I believe the connect() call should bind the socket for you.